_posts/2017-08-25-introducing-sandboxfs.md

```diff
@@ -3,40 +3,46 @@ layout: posts
 title: Introducing sandboxfs
 ---
 
+sandboxfs is a new sandboxing technique for Bazel. It's experimental and
+subject to change, but it's available now for you to check out!
+
+# Correct builds
+
 As our motto claims, *correctness* is an integral part of Bazel. To achieve
 correctness, builds must be *hermetic* and
 [*reproducible*](https://reproducible-builds.org/), which means that all
 actions invoked by Bazel should be run in a well-defined environment. In other
 words: we want actions to run within a sandbox.
 
-To better illustrate this point, let's consider an example: think of a C++
-`cc_library` target that specifies `foo.cc` as a source. This target contains a
+Let's consider an example. Think of a `cc_library` target that specifies
+`foo.cc` as a source. The `cc_library` rule contains a
 *compilation action* that turns the `foo.cc` source file into a `foo.o` object
-file. This action will run clang, will need to read `foo.cc`, will need to read
-the transitive closure of all headers required by `foo.cc`, will need to write
-`foo.o`, and may need to write some temporary files under `/tmp/`.
-
-Under these conditions, Bazel must ensure that the action only has read-only
-access to the input files we identified and write-only access to the output and
-temporary files we expect. The reason for this is because, unless these
-conditions are enforced, Bazel cannot know that the output file really came
-from those inputs: it may well be that the compiler went astray and read random
-files from the system, which means further builds would be inconsistent.
+file. This action will run clang, and it will need to:
+*  read `foo.cc`,
+*  read the transitive closure of all headers required by `foo.cc`
+*  write `foo.o`, and
+*  possibly write some temporary files under `/tmp/`.
+
+To be confident this build is correct, Bazel must ensure that the action has
+read-only access to the identified input files and write-only access
+to the output and temporary files we expect. Otherwise, Bazel cannot know if the
```

Comment:
enforcing the correct read only and write only == sandboxing?

We could add "Ensuring these conditions (or restrictions?) is sandbox." or "Sandboxing ensures these conditions."

Reply:
Added something along these lines.

Reply:
looks good.